questionpy_sdk/webserver/static/conditions.js

```diff
@@ -0,0 +1,120 @@
+export class Types {
+    constructor(value) {
+        this.value = value
+    }
+
+    static values = {
+        hide_if: 'hide_if',
+        disable_if: 'disable_if'
+    }
+
+    to_selector() {
+        return "[".concat(this.value, "]");
+    }
+}
+
+
+export class Condition {
+    constructor(kind, name) {
+        if (this.constructor === Condition) {
+            throw new Error("Cannot create instance of abstract class")
+        }
+        this.kind = kind;
+        this.name = name;
+        this.targets = this.set_targets();
+    }
+
+    /**
+     * Create a Condition from a plain javascript object. The Condition type is chosen according to the
+     * object kind.
+     *
+     * Ex.: object.kind = "is_checked" returns new IsChecked(object.name)
+     * @param object the plain javascript object
+     * @returns {Condition} the condition object
+     */
+    static from_object(object) {
+        if (!object || !object.kind) {
+            throw new Error("Invalid condition")
+        }
+        switch (object.kind) {
+            case "is_checked":
+                return new IsChecked(object.name);
+            case "is_not_checked":
+                return new IsNotChecked(object.name);
+            case "equals":
+                return new Equals(object.name, object.value);
+            case "does_not_equal":
+                return new DoesNotEqual(object.name, object.value);
+            case "in":
+                return new In(object.name, object.value);
+            default:
+                throw new Error("Invalid condition kind")
+        }
+    }
+
+    set_targets() {
+        return Array.from(document.querySelectorAll("#".concat(this.name)));
```

Comment:
* Der CSS-Selector `#abc` sucht nach einem Element mit der ID `abc`. Da eine ID in einem HTML-Dokument höchstens einmal vergeben sein sollte, muss das kein Array sein.
* Hier braucht es gar keinen CSS-Selektor. `document.getElementById` reicht (und gibt auch nur ein Element oder null zurück)
* Die Methode setzt nichts. Außerdem wird sie nur im Konstruktor benutzt. Da braucht es vermutlich gar keine Methode

Reply:
Guter Punkt. Da habe ich auch gleich einen zweiten Fehler gefunden, dass `conditions` nur ein `target` haben können.